### PR TITLE
Implement ConsensusOutputAPI for mysticeti

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

Implement ConsensusOutputAPI for CommittedSubDag in Mysticeti.
Instead of displaying the digest of consensus output, we now rely on the Display impl of the consensus output.
Also changed into_transactions() to transactions(), instead, which returns reference to the serialized bytes to avoid clones.
Left a few TODOs in the impl

## Test Plan 

CI

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-core/src/consensus_types/consensus_output_api.rs [hunk 1]
```diff
@@ -89,3 +90,59 @@ impl ConsensusOutputAPI for narwhal_types::ConsensusOutput {
...
+        self.anchor.round
+    }
+
+    fn leader_author_index(&self) -> AuthorityIndex {
+        self.anchor.authority as AuthorityIndex
+    }
+
+    fn commit_timestamp_ms(&self) -> u64 {
+        // TODO: Enforce ordered timestamp in Mysticeti
+        0
```
Comment (andll) [inline crates/sui-core/src/consensus_types/consensus_output_api.rs:110]: We can still return timestamp from the leader block for now (I am not sure if anything can break in sui if timestamp is always 0)

File: crates/sui-core/src/consensus_types/consensus_output_api.rs [hunk 2]
```diff
@@ -89,3 +90,59 @@ impl ConsensusOutputAPI for narwhal_types::ConsensusOutput {
...
+            .into_iter()
+            .map(|block| {
+                let round = block.round();
+                let author = block.author() as AuthorityIndex;
+                let transactions: Vec<_> = block
+                    .statements()
+                    .iter()
+                    .enumerate()
+                    .flat_map(|(idx, stmt)| match stmt {
+                        BaseStatement::Share(tx) => {
```
Comment (andll) [inline crates/sui-core/src/consensus_types/consensus_output_api.rs:129]: You can also use `StatementBlock::shared_transactions` which gives you iterator of transactions

**Review by andll — APPROVED** [review]

File: crates/sui-core/src/consensus_types/consensus_output_api.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-core/src/consensus_types/consensus_output_api.rs:110]: > timestamp from the leader block for now

But that's not enforced to be monotonic, right? That would be worse than returning 0?

Comment (andll) [reply crates/sui-core/src/consensus_types/consensus_output_api.rs:110]: It's not enforced, but I think it is very unlikely

**Review by akichidis — APPROVED** [review]

**Pull request merged**